testbed/deploy_resources.go

```diff
@@ -0,0 +1,197 @@
+package testbed
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/common/aws"
+	test_utils "github.com/Layr-Labs/eigenda/common/aws/dynamodb/utils"
+	"github.com/Layr-Labs/eigenda/common/store"
+	"github.com/Layr-Labs/eigenda/core/meterer"
+	"github.com/Layr-Labs/eigenda/disperser/common/blobstore"
+	blobstorev2 "github.com/Layr-Labs/eigenda/disperser/common/v2/blobstore"
+	awssdk "github.com/aws/aws-sdk-go-v2/aws"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+// DeployResourcesConfig holds configuration for deploying AWS resources
+type DeployResourcesConfig struct {
+	LocalStackEndpoint  string
+	MetadataTableName   string
+	BucketTableName     string
+	V2MetadataTableName string
+	V2PaymentPrefix     string // Optional: prefix for v2 payment tables, defaults to "e2e_v2_"
+	Region              string // Optional: AWS region, defaults to "us-east-1"
+	AccessKey           string // Optional: AWS access key, defaults to "localstack"
+	SecretAccessKey     string // Optional: AWS secret key, defaults to "localstack"
+}
+
+// DeployResources creates AWS resources (S3 buckets and DynamoDB tables) on LocalStack
+func DeployResources(ctx context.Context, config DeployResourcesConfig) error {
+	// Set defaults
+	if config.Region == "" {
+		config.Region = "us-east-1"
+	}
+	if config.AccessKey == "" {
+		config.AccessKey = "localstack"
+	}
+	if config.SecretAccessKey == "" {
+		config.SecretAccessKey = "localstack"
+	}
+	if config.V2PaymentPrefix == "" {
+		config.V2PaymentPrefix = "e2e_v2_"
+	}
+
+	// Create AWS client config
+	cfg := aws.ClientConfig{
+		Region:          config.Region,
+		AccessKey:       config.AccessKey,
+		SecretAccessKey: config.SecretAccessKey,
+		EndpointURL:     config.LocalStackEndpoint,
+	}
+
+	// Create S3 bucket
+	if err := createS3Bucket(ctx, cfg); err != nil {
+		return fmt.Errorf("failed to create S3 bucket: %w", err)
+	}
+
+	// Create metadata table
+	if config.MetadataTableName != "" {
+		_, err := test_utils.CreateTable(ctx, cfg, config.MetadataTableName,
+			blobstore.GenerateTableSchema(config.MetadataTableName, 10, 10))
+		if err != nil {
+			return fmt.Errorf("failed to create metadata table %s: %w", config.MetadataTableName, err)
+		}
+		fmt.Printf("Created metadata table: %s\n", config.MetadataTableName)
```

Comment:
Would it be possible to use a logger instead of `fmt.Printf` in this file? If you decide to accept this suggestion, then I'd suggest searching for all prints in this PR and replacing them (not going to comment each location 😅).